bigquery/tests/unit/test__pandas_helpers.py

```diff
@@ -511,9 +514,175 @@ def test_bq_to_arrow_schema_w_unknown_type(module_under_test):
+    expected = pandas.Series([1, 2, 3], name="column_name")
+    pandas.testing.assert_series_equal(series, expected)
+
+
+@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
+def test_get_column_or_index_with_index(module_under_test):
+    dataframe = pandas.DataFrame(
+        {"column_name": [1, 2, 3]}, index=pandas.Index([4, 5, 6], name="index_name")
+    )
+    series = module_under_test.get_column_or_index(dataframe, "index_name")
+    expected = pandas.Series([4, 5, 6], name="index_name")
+    pandas.testing.assert_series_equal(series, expected)
+
+
+@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
+def test_get_column_or_index_with_datetimeindex(module_under_test):
+    datetimes = [
+        datetime.datetime(2000, 1, 2, 3, 4, 5, 101),
+        datetime.datetime(2006, 7, 8, 9, 10, 11, 202),
+        datetime.datetime(2012, 1, 14, 15, 16, 17, 303),
+    ]
+    dataframe = pandas.DataFrame(
+        {"column_name": [1, 2, 3]},
+        index=pandas.DatetimeIndex(datetimes, name="index_name"),
+    )
+    series = module_under_test.get_column_or_index(dataframe, "index_name")
+    expected = pandas.Series(datetimes, name="index_name")
+    pandas.testing.assert_series_equal(series, expected)
+
+
+@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
+def test_get_column_or_index_with_multiindex(module_under_test):
+    dataframe = pandas.DataFrame(
+        {"column_name": [1, 2, 3, 4, 5, 6]},
+        index=pandas.MultiIndex.from_tuples(
+            [("a", 0), ("a", 1), ("b", 0), ("b", 1), ("c", 0), ("c", 1)],
+            names=["letters", "numbers"],
+        ),
+    )
+
+    series = module_under_test.get_column_or_index(dataframe, "letters")
+    expected = pandas.Series(["a", "a", "b", "b", "c", "c"], name="letters")
+    pandas.testing.assert_series_equal(series, expected)
+
+    series = module_under_test.get_column_or_index(dataframe, "numbers")
+    expected = pandas.Series([0, 1, 0, 1, 0, 1], name="numbers")
+    pandas.testing.assert_series_equal(series, expected)
+
+
+@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
+def test_list_columns_and_indexes_without_named_index(module_under_test):
+    df_data = collections.OrderedDict(
+        [
+            ("a_series", [1, 2, 3, 4]),
+            ("b_series", [0.1, 0.2, 0.3, 0.4]),
+            ("c_series", ["a", "b", "c", "d"]),
+        ]
+    )
+    dataframe = pandas.DataFrame(df_data)
+
+    columns_and_indexes = module_under_test.list_columns_and_indexes(dataframe)
+    expected = [
+        ("a_series", pandas.api.types.pandas_dtype("int64")),
+        ("b_series", pandas.api.types.pandas_dtype("float64")),
+        ("c_series", pandas.api.types.pandas_dtype("object")),
+    ]
+    assert columns_and_indexes == expected
+
+
+@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
+def test_list_columns_and_indexes_with_index(module_under_test):
```

Comment:
(nit)
Maybe rename to `..._with_named_index` to emphasize the "named" test case?